testsuite/replay-verify/main.py

```diff
@@ -0,0 +1,298 @@
+
+    def get_pod_status(self):
+        pod_status = self.client.read_namespaced_pod_status(
+            name=self.name, namespace=self.namespace
+        )
+        return pod_status
+
+    def get_humio_log_link(self):
+        # TODO: Implement this ref:get_humio_link_for_node_logs
+        return f"https://humio.com/search?query=namespace%3D%22{self.namespace}%22%20pod%3D%22{self.name}%22"
+
+
+class ReplayScheduler:
+    def __init__(
+        self,
+        id,
+        start_version,
+        end_version,
+        ranges_to_skip,
+        worker_cnt=NUM_OF_WORKERS,
+        range_size=RANGE_SIZE,
+        network=Network.TESETNET,
+    ):
+        config.load_kube_config()
+        self.client = client.CoreV1Api()
+        self.id = id
+        self.namespace = "default"
+        self.start_version = start_version
+        self.end_version = end_version
+        self.ranges_to_skip = ranges_to_skip
+        self.range_size = range_size
+        self.ranges_to_skip = ranges_to_skip
+        self.current_workers = [None] * worker_cnt
+        self.tasks = []
+        self.network = network
+        self.failed_workpod_logs = []
+        self.txn_mismatch_logs = []
+
+    def get_label(self):
+        return f"{self.id}-{self.network}"
+
+    # Create ranges consisting of start_version and end_version
+    def create_ranges(self):
+        ranges = []
+        current = self.start_version
+
+        # Sort ranges_to_skip to ensure we process them in order
+        sorted_skips = sorted(self.ranges_to_skip, key=lambda x: x[0])
+
+        while current < self.end_version:
+            # Check if current position falls within any skip range
+            should_skip = False
+            for skip_start, skip_end in sorted_skips:
+                if skip_start <= current < skip_end:
+                    current = skip_end  # Jump to end of skip range
+                    should_skip = True
+                    break
+
+            if not should_skip:
+                range_end = min(current + self.range_size, self.end_version)
+                ranges.append((current, range_end))
+                current += self.range_size
+
+        self.tasks = ranges
+
+    def schedule(self, from_scratch=False):
+        if from_scratch:
+            self.kill_all_running_pods(self.get_label())
+        self.create_ranges()
+
+        while len(self.tasks) > 0:
+            for i in range(len(self.current_workers)):
+                if (
+                    self.current_workers[i] is None
+                    or self.current_workers[i].is_completed()
+                ):
+                    if (
+                        self.current_workers[i] is not None
+                        and self.current_workers[i].is_completed()
+                    ):
+                        self.collect_logs_from_completed_pods(self.current_workers[i])
+                    if len(self.tasks) == 0:
+                        break
+                    chunk = self.tasks.pop(0)
```

Comment:
nit: you are using three concepts "range", "task" and "chunk" to refer to exactly the same thing :trollface: